core/new-gui/src/app/workspace/component/visualization-panel-content/visualization-panel-content.component.ts

```diff
@@ -102,6 +119,95 @@ export class VisualizationPanelContentComponent implements AfterViewInit, OnDest
       case ChartType.SPLINE:
         this.generateChart();
         break;
+      case ChartType.SPATIAL_SCATTERPLOT:
+        this.generateSpatialScatterplot();
+        break;
+      case ChartType.SCATTERPLOT:
+        this.generateSimpleScatterplot();
+        break;
+    }
+  }
+
+  generateSimpleScatterplot() {
+    if (this.c3ChartElement) {
+      this.c3ChartElement.destroy();
+    }
+    const result = this.data as Array<Record<string, Primitive>>;
+    const xLabel: string = Object.keys(result[0])[0];
+    const yLabel: string = Object.keys(result[0])[1];
+
+    this.c3ChartElement = c3.generate({
+      size: {
+        height: VisualizationPanelContentComponent.HEIGHT,
+        width: VisualizationPanelContentComponent.WIDTH
+      },
+      data: {
+        json: result,
+        keys: {
+          x: xLabel,
+          value: [yLabel]
+        },
+        type: this.chartType as c3.ChartType
+      },
+      axis: {
+        x: {
+          label: xLabel,
+          tick: {
+            fit: true
+          }
+        },
+        y: {
+          label: yLabel
+        }
+      },
+      bindto: VisualizationPanelContentComponent.CHART_ID
+    });
+  }
+
+  generateSpatialScatterplot() {
+    if (this.map === undefined) {
+      this.initMap();
+    }
+    this.map?.on('styledata', () => {
+      this.addScatterLayer();
+    });
+  }
+
+  initMap() {
+    /* mapbox object with default configuration */
+    this.map = new mapboxgl.Map({
+      container: VisualizationPanelContentComponent.MAP_CONTAINER,
+      style: 'mapbox://styles/mapbox/light-v9',
+      center: [-96.35, 39.5],
+      zoom: 3,
+      maxZoom: 17,
+      minZoom: 0
+    });
+  }
+
+  addScatterLayer() {
+    if (this.map !== undefined) {
```

Comment:
should at least log an error if `map` is not defined. the method should not succeed when map is undefined.

Reply:
this function won't be even called if map is not defined, I was just making the compiler happy

Reply:
Still, the method has an unhandled case when the map is null. It will be good to handle all if-branches explicitly IMO.

Reply:
that is handled before even calling the function

Reply:
@sadeemsaleh  one trick we do to make the compiler happy is to return early if the value is undefined/null, like this:
```
if (! this.map) {
    return;
}
// then you can use this variable normally
this.map // compiler now knows that this is not undefined/empty
```